moralization/spacy_model.py

```diff
@@ -0,0 +1,271 @@
+        :type data_dir: str/Path
+        :param working_dir: Directory where the training data, configs and results are stored., defaults to Debug value
+        :type working_dir: _type_, optional
+        :param config_file: Filename or path for the config file, defaults to searching in the working directory.
+        :type config_file: _type_, optional
+        """
+
+        self.data_dir, self.working_dir = self._setup_working_dir(data_dir, working_dir)
+
+    def _setup_working_dir(self, data_dir, working_dir):
+
+        # maybe set default working_dir to tmp dir
+        data_dir = Path(data_dir)
+
+        if working_dir:
+            working_dir = Path(working_dir)
+        else:
+            working_dir = Path(mkdtemp())
+
+        pathlib.Path(working_dir).mkdir(exist_ok=True)
+        return data_dir, working_dir
+
+    def convert_data_to_spacy_doc(self):
+        """convert the given xmi/xml files to a spacy specific binary filesystem.
+
+        :param output_dir: where to store generated files. If None is given the working dir will be used
+        :type output_dir: dir
+        """
+        data_dict = InputOutput.read_data(self.data_dir)
+
+        self.doc_dict = self._convert_dict_to_doc_dict(data_dict)
+
+        return self.doc_dict
+
+    def export_training_testing_data(self, output_dir=None):
+        """_summary_
+
+        :param output_dir: _description_, defaults to None
+        :type output_dir: _type_, optional
+        """
+        if output_dir is None:
+            output_dir = self.working_dir
+
+        db_train = DocBin()
+        db_dev = DocBin()
+
+        for file in self.doc_dict.values():
+            db_train.add(file["train"])
+            db_dev.add(file["dev"])
+
+        db_train.to_disk(output_dir / "train.spacy")
+        db_dev.to_disk(output_dir / "dev.spacy")
+
+    def visualize_data(self, filename=None, type="all", style="span"):
+        """Use the displacy class offered by spacy to visualize the current dataset.
+
+        :param filename:    Specify which of the loaded files should be presented, if None all files are shown.
+                            This can also take a list., defaults to None
+        :type filename: str/list, optional
+        :param type: Specify is only the trainings, the testing or all datapoints should be shown, defaults to "all"
+        :type type: str, optional
+        :param type: the visualization type given to displacy, available are "dep", "ent" and "span, defaults to "span".
+
+        """
+        if not is_interactive():
+            raise NotImplementedError(
+                "Please only use this function in a jupyter notebook for the time being."
+            )
+
+        if filename is None:
+            filename = list(self.doc_dict.keys())
+        return displacy.render(
+            [self.doc_dict[file][type] for file in filename], style=style
+        )
+
+    def _convert_dict_to_doc_dict(self, data_dict):
```

Comment:
We will leave it as is for now, as the ML modules are being extended there will anyway be refactoring of this at a later stage. It is quite readable despite the nested statements.